File: backend/app/schemas/growth.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any

class GrowthBasketItemDTO(BaseModel):
    product_id: str
    name: str
    brand: Optional[str] = "RazorRecon Commerce"
    category: str = "General"
    price: float
    cost_price: Optional[float] = None
    quantity: int = 1
    image_url: Optional[str] = "https://images.unsplash.com/photo-1556742049-0a67e557224f?w=500"

class GrowthBasketRequestDTO(BaseModel):
    items: List[GrowthBasketItemDTO] = []
    merchant_segment: Optional[str] = "Retail & D2C E-Commerce"

class RecommendationCardDTO(BaseModel):
    id: str
    type: str # "upsell" | "cross_sell"
    title: str
    badge_label: str
    target_product_id: str
    target_product_name: str
    target_brand: str
    target_category: str
    target_image_url: str
    target_price: float
    target_cost_price: float
    original_product_id: Optional[str] = None
    original_product_name: Optional[str] = None
    price_delta: float
    margin_delta_pct: float
    confidence_score_pct: int
    conversion_probability: float
    expected_uplift_inr: float
    strategy_rationale: str
    key_advantages: List[str] = []

class AffinityRuleDTO(BaseModel):
    rule_id: str
    antecedent_product_name: str
    consequent_product_name: str
    consequent_product_id: str
    consequent_price: float
    support_pct: float
    confidence_pct: float
    lift_score: float
    historical_co_purchases: int
    synergy_type: str # "Hardware + Audio" | "POS + Software" | "Peripheral + Warranty"

class SampleBasketDTO(BaseModel):
    id: str
    name: str
    description: str
    industry: str
    items: List[GrowthBasketItemDTO]

class GrowthAnalysisResponseDTO(BaseModel):
    current_cart_value: float
    predicted_cart_value: float
    expected_uplift_pct: float
    expected_uplift_inr: float
    current_gross_margin_pct: float
    projected_gross_margin_pct: float
    margin_expansion_pct: float
    total_active_items: int
    upsell_recommendations: List[RecommendationCardDTO] = []
    cross_sell_recommendations: List[RecommendationCardDTO] = []
    affinity_rules: List[AffinityRuleDTO] = []
    ai_strategy_rationale: str
    growth_health_score: int # 0-100

File: frontend/src/types/growth.ts
export interface GrowthBasketItem {
  product_id: string;
  name: string;
  brand: string;
  category: string;
  price: number;
  cost_price: number;
  quantity: number;
  image_url: string;
}

export interface RecommendationCard {
  id: string;
  type: 'upsell' | 'cross_sell';
  title: string;
  badge_label: string;
  target_product_id: string;
  target_product_name: string;
  target_brand: string;
  target_category: string;
  target_image_url: string;
  target_price: number;
  target_cost_price: number;
  original_product_id?: string;
  original_product_name?: string;
  price_delta: number;
  margin_delta_pct: number;
  confidence_score_pct: number;
  conversion_probability: number;
  expected_uplift_inr: number;
  strategy_rationale: string;
  key_advantages: string[];
}

export interface AffinityRule {
  rule_id: string;
  antecedent_product_name: string;
  consequent_product_name: string;
  consequent_product_id: string;
  consequent_price: number;
  support_pct: number;
  confidence_pct: number;
  lift_score: number;
  historical_co_purchases: number;
  synergy_type: string;
}

export interface SampleBasket {
  id: string;
  name: string;
  description: string;
  industry: string;
  items: GrowthBasketItem[];
}

export interface GrowthAnalysisResponse {
  current_cart_value: number;
  predicted_cart_value: number;
  expected_uplift_pct: number;
  expected_uplift_inr: number;
  current_gross_margin_pct: number;
  projected_gross_margin_pct: number;
  margin_expansion_pct: number;
  total_active_items: number;
  upsell_recommendations: RecommendationCard[];
  cross_sell_recommendations: RecommendationCard[];
  affinity_rules: AffinityRule[];
  ai_strategy_rationale: string;
  growth_health_score: number;
}
